src/charm.py

```diff
@@ -64,6 +68,28 @@ def _on_config_changed(self, event) -> None:
         self.model.unit.status = MaintenanceStatus(msg)
         self.configure_workload_container(event)
 
+    def _report_visits_by_ip_action(self, event: ActionEvent) -> None:
+        """Handle the report-visits-by-ip action."""
+        results = self._report_visits_by_ip()
+        event.set_results({"ips": str(results)})
+
+    def _report_visits_by_ip(self) -> dict[str, int]:
```

Comment:
`Dict` from `typing`instead of `dict` maybe?

Reply:
I am using a python dict, not a typeDict (simpler implementation, since Python does not enforce type hints anyway)

Reply:
I'm referring to the type hint, it should probably be `Dict[str, int]`. See https://stackoverflow.com/questions/37087457/difference-between-defining-typing-dict-and-dict